src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotifyClaimCallbackHandler.java

```diff
@@ -100,4 +155,29 @@ private SubmittedCallbackResponse buildConfirmation(CallbackParams callbackParam
             .confirmationBody(body)
             .build();
     }
+
+    private SubmittedCallbackResponse buildConfirmationWithSolicitorOptions(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        String formattedDeadline = formatLocalDateTime(caseData.getClaimDetailsNotificationDeadline(), DATE_TIME_AT);
+        String body;
+
+        if (notifyBothRespondentSolicitors(caseData)) {
+            body = format(CONFIRMATION_SUMMARY, formattedDeadline) + exitSurveyContentService.applicantSurvey();
+        } else {
+            body = format(CONFIRMATION_NOTIFICATION_ONE_PARTY_SUMMARY, formattedDeadline)
+                + exitSurveyContentService.applicantSurvey();
+        }
+
+        return SubmittedCallbackResponse.builder()
+            .confirmationHeader(String.format(
+                "# Notification of claim sent%n## Claim number: %s",
+                caseData.getLegacyCaseReference()
+            ))
+            .confirmationBody(body)
+            .build();
+    }
+
+    protected boolean notifyBothRespondentSolicitors(CaseData caseData) {
+        return Objects.equals("Both", caseData.getDefendantSolicitorNotifyClaimOptions().getValue().getLabel());
```

Comment:
This also needs to be made backward compatible.

I suspect that as it stands, this won't work if you merge the service code first and only the service code (that is without the accompanying CCD Def changes)

Reply:
This code will be evaluated by the mid event, so I have included a multiparty as part of that logic also:
```
        if (featureToggleService.isMultipartyEnabled()
            && !notifyBothRespondentSolicitors(caseData)) {
            warnings.add(WARNING_ONLY_NOTIFY_ONE_DEFENDANT_SOLICITOR);
        }
```
So warnings will only be evaluated when Multiparty is on